openedx/features/philu_courseware/helpers.py

```diff
@@ -87,4 +86,27 @@ def get_pre_post_assessments_score(user, course_id, chapter_id):
         'pre_assessment_score': pre_assessment_score,
         'post_assessment_score': post_assessment_score,
         'pre_assessment_attempted': pre_assessment_attempted,
+        'all_pre_assessment_attempted': all_pre_assessment_attempted,
+        'all_post_assessment_attempted': all_post_assessment_attempted,
     }
+
+
+def is_all_attempted(section):
+    attempted_problem_scores = [score for score in section.problem_scores.values()
+                                    if score.first_attempted] \
+                                    if section and section.problem_scores else []
+    problems_count = configuration_helpers.get_value("COMPETENCY_ASSESSMENT_PROBLEMS_COUNT", 
+                        constants.COMPETENCY_ASSESSMENT_DEFAULT_PROBLEMS_COUNT)
+    return len(attempted_problem_scores) == problems_count
+
+
+def is_pre_assessment(section):
+    return get_section_format(section) == constants.PRE_ASSESSMENT_FORMAT
+
+
+def is_post_assessment(section):
+    return get_section_format(section) == constants.POST_ASSESSMENT_FORMAT
+
+
+def get_section_format(section):
+    return section.format.lower() if section and section.format else ''
```

Comment:
The sanity check `if section` has two occurrence, one in  `get_section_format` other in `is_all_attempted`, both functions are executing sequentially, I think this sanity should be at one place @zeshanarifios